src/pages/home/report/ListBoundaryLoader/ListBoundaryLoader.js

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import ReportActionsSkeletonView from '../../../../components/ReportActionsSkeletonView';
+import CONST from '../../../../CONST';
+import useNetwork from '../../../../hooks/useNetwork';
+import ListHeaderComponentLoader from './ListHeaderComponentLoader/ListHeaderComponentLoader';
+
+const propTypes = {
+    type: PropTypes.string.isRequired,
+    isLoadingOlderReportActions: PropTypes.bool,
+    isLoadingInitialReportActions: PropTypes.bool,
+    isLoadingNewerReportActions: PropTypes.bool,
+    skeletonViewHeight: PropTypes.number,
+    lastReportActionName: PropTypes.string,
+};
+
+const defaultProps = {
+    isLoadingOlderReportActions: false,
+    isLoadingInitialReportActions: false,
+    isLoadingNewerReportActions: false,
+    skeletonViewHeight: 0,
+    lastReportActionName: '',
+};
+
+function ListBoundaryLoader({type, isLoadingOlderReportActions, isLoadingInitialReportActions, skeletonViewHeight, lastReportActionName, isLoadingNewerReportActions}) {
+    const {isOffline} = useNetwork();
+    // we use two different loading components for header and footer to reduce the jumping effect when you scrolling to the newer reports
+    if (type === CONST.LIST_COMPONENTS.FOOTER) {
+        if (isLoadingOlderReportActions) {
+            return <ReportActionsSkeletonView containerHeight={CONST.CHAT_SKELETON_VIEW.AVERAGE_ROW_HEIGHT * 3} />;
+        }
+
+        // Make sure the oldest report action loaded is not the first. This is so we do not show the
+        // skeleton view above the created action in a newly generated optimistic chat or one with not
+        // that many comments.
+        // const lastReportAction = _.last(sortedReportActions) || {};
+        if (isLoadingInitialReportActions && lastReportActionName !== CONST.REPORT.ACTIONS.TYPE.CREATED) {
+            return (
+                <ReportActionsSkeletonView
+                    containerHeight={skeletonViewHeight}
+                    animate={!isOffline}
```

Comment:
NAB but this prop should be called `shouldAnimate`:

```suggestion
                    shouldAnimate={!isOffline}
```

Reply:
👍🏻